vcd/writer.py

```diff
@@ -79,6 +79,7 @@ def __init__(self, file, timescale='1 us', date=None, comment='',
         self._scope_var_names = {}
         self._scope_types = {}
         self._ident_values = OrderedDict()
+        self._init_timestamp = init_timestamp
         self._prev_timestamp = 0
```

Comment:
Could the problem have been with the `self._dump_off(0)` on line 411? That hardwired `0` is a timestamp.

Other than that, I do not see assumptions about registration being at time 0.

Even if we find that we have to keep `_init_timestamp`, that `_dump_off(0)` call needs to be addressed.